test/router-luatest/router_test.lua

```diff
@@ -545,3 +545,44 @@ g.test_enable_disable = function(g)
     -- we don't want this server to interfere with subsequent tests
     g.router_1:drop()
 end
+
+g.test_explicit_fiber_kill = function(g)
+    --
+    -- Kill fibers and wait until they're dead. Without waiting the fiber can
+    -- die during request, which will cause an exception to be thrown.
+    --
+    local rs_uuids = {g.replica_1_a:replicaset_uuid(),
+                      g.replica_2_a:replicaset_uuid()}
+
+    g.router:exec(function(uuids)
+        for id, f in pairs(ifiber.info()) do
+            if f.name:endswith('(net.box)') then
+                ifiber.kill(id)
+            end
+        end
```

Comment:
I am afraid we might accidentally kill something important which is not a part of vshard and won't have that kind of protection against dead netbox fibers. For example, we could damage the always fragile luatest. I would recommend to kill just the needed fibers. You can do it like that:
```Lua
conn:eval([[
    box.session.push(nil)
]], {}, {on_push = function()
    ifiber.self():cancel()
end})
```
Here I sent a request which returns 2 responses - one "push" and the final result (which is nil here). For pushes we can install callbacks which are executed in the same fiber as netbox. It allows you to get to the worker fiber and kill it directly.

Reply:
Great hack